runtime/caml/blake2.h

```diff
@@ -0,0 +1,46 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*           Xavier Leroy, Collège de France and Inria Paris              */
+/*                                                                        */
+/*   Copyright 2022 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+/* BLAKE2 message digest */
+
+#ifndef CAML_BLAKE2_H
+#define CAML_BLAKE2_H
+
+#ifdef CAML_INTERNALS
+
+#include "mlvalues.h"
+
+#define BLAKE2_BLOCKSIZE 128
+
+struct BLAKE2_context {
+  uint64_t h[8];
+  uint64_t len[2];
+  int numbytes;
+  unsigned char buffer[BLAKE2_BLOCKSIZE];
+};
+
+CAMLextern void
+caml_BLAKE2Init(struct BLAKE2_context * context,
+                int hashlen, int keylen, const unsigned char * key);
```

Comment:
Here also, do we want to make `hashlen` unsigned?

Reply:
OK for unsigned types, see 6d44ff256342e4b9c3973f88703d75fa5e9fea2f